src/python/pants/engine/internals/parser.py

```diff
@@ -70,17 +70,19 @@ def __call__(self, *args, **kwargs):
                 self._parse_context._storage.add(obj)
                 return obj
 
-        for alias, symbol in symbol_table.table.items():
-            registrar = Registrar(parse_context, alias, object_type=symbol)
-            symbols[alias] = registrar
-            symbols[symbol] = registrar
-
-        if aliases.objects:
-            symbols.update(aliases.objects)
-
-        for alias, object_factory in aliases.context_aware_object_factories.items():
-            symbols[alias] = object_factory(parse_context)
-
+        symbols.update(
+            {
+                alias: Registrar(parse_context, alias, object_type=symbol)
+                for alias, symbol in symbol_table.table.items()
+            }
+        )
+        symbols.update(aliases.objects)
```

Comment:
Are we confident that `aliases.objects is never None?

Reply:
Yes. Both MyPy + the constructor validate that it is always a `FrozenDict`.